libsolidity/interface/CompilerStack.cpp

```diff
@@ -1329,7 +1329,7 @@ void CompilerStack::generateIR(ContractDefinition const& _contract)
 	for (auto const& pair: m_contracts)
 		otherYulSources.emplace(pair.second.contract, pair.second.yulIR);
 
-	IRGenerator generator(m_evmVersion, m_revertStrings, m_optimiserSettings, sourceIndices());
+	IRGenerator generator(m_evmVersion, m_revertStrings, m_optimiserSettings, sourceIndices(), this);
```

Comment:
This now passes the `CompilerStack` instance (as `CharStreamProvider`) all the way down through `IRGenerator`, `IRGeneratorContext`, `AssemblyStack` and `Object` down to `AsmPrinter`. All the methods on the way need the provider as an optional argument. Not sure if there's a way to make it nicer.This now passes the `CompilerStack` instance (as `CharStreamProvider`) all the way down through `IRGenerator`, `IRGeneratorContext`, `AssemblyStack` and `Object` down to `AsmPrinter`. All the methods on the way need the provider as an optional argument. Not sure if there's a way to make it nicer.